app/components/badge-github-actions.js

```diff
@@ -0,0 +1,61 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { alias } from '@ember/object/computed';
+
+export default Component.extend({
+  tagName: 'span',
+  repository: alias('badge.attributes.repository'),
+  imageUrl: computed('badge.attributes.{repository,workflow_enc,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/workflows/${this.workflow_enc}/badge.svg`);
+
+    if (this.branch !== '') {
+      url.searchParams.set('branch', this.branch);
+    }
+
+    if (this.event !== '') {
+      url.searchParams.set('event', this.event);
+    }
+
+    return url.href;
+  }),
+  url: computed('badge.attributes.{repository,workflow,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/actions`);
+
+    let query = '';
+    if (this.workflow !== '') {
+      query += `workflow:"${this.workflow}"`;
```

Comment:
For anyone who might have been wondering: quotes can be escaped as `\"`, but there doesn’t seem to be a way to represent a query where multiple quoted terms end with a backslash. Because out of these three, only workflow names can end with backslashes, it *is* possible to support every workflow name by:

- escaping double quotes
- putting the `workflow:` part last

I’m sure trailing backslashes aren’t considered important enough to support here, but how about ASCII double quotes in workflow and branch names?